sdk/eventhub/event-hubs/src/partitionPump.ts

```diff
@@ -32,16 +32,24 @@ export class PartitionPump {
     this._abortController = new AbortController();
   }
 
-  async start(partitionId: string): Promise<void> {
-    if (this._partitionProcessor.initialize) {
-      await this._partitionProcessor.initialize();
+  public get isReceiving(): boolean {
+    return this._isReceiving;
+  }
+
+  async start(): Promise<void> {
+    this._isReceiving = true;
+    if (typeof this._partitionProcessor.initialize === "function") {
+      try {
+        await this._partitionProcessor.initialize();
+      } catch {
+        this._isReceiving = false;
```

Comment:
What is the expectation for when initialize fails?
Here we swallow the error and move on.
If so, then why is _isReceiving set to false if we call `recieveEvents` in the next line

Reply:
Ah, I think I meant to close the partitionPump, but I'm not sure what the right thing to do here is...expect the user to handle all errors in their own code and keep going or treat a thrown error as meaning STOP.

Either way what I have now isn't right, I'll reach out to other teams to see how they plan to handle errors thrown by PartitionProcessor.

Reply:
For now I decided to swallow the error but we can revisit.